src/mesido/potential_errors.py

```diff
@@ -0,0 +1,65 @@
+from typing import Dict, List
+
+from mesido.exceptions import MesidoAssetIssue
+from mesido.network_common import MesidoAssetIssueType
+
+AssetId = str
+ErrorMessage = str
+
+
+class PotentialErrors:
+    """Singleton, do not instantiate. Use POTENTIAL_ERRORS."""
+
+    _gathered_potential_issues: Dict[MesidoAssetIssueType, Dict[AssetId, List[ErrorMessage]]]
+
+    def __init__(self):
+        self._gathered_potential_issues = {}
+
+    def add_potential_issue(
+        self, issue_type: MesidoAssetIssueType, asset_id: AssetId, error_message: ErrorMessage
+    ) -> None:
+        """
+        Add potential issues to _gathered_potential_issues.
+
+        """
+        self._gathered_potential_issues.setdefault(issue_type, {}).setdefault(asset_id, "")
+        self._gathered_potential_issues[issue_type][asset_id] = error_message
+
+    def have_issues_for(self, issue_types: List[MesidoAssetIssueType]) -> Dict[
+        MesidoAssetIssueType,
+        bool,
+    ]:
+        """
+        Check if the potential issue exists and mark them as True for the specific issue type.
+
+        """
+        result = {}
+        for ii in range(len(issue_types)):
```

Comment:
Can't this be changed in a loop without indexing and where you take the issue type directly from the list, e.g.:
for i_type in issue_types:
then there is no need for the issue_types[ii] but you can immediately use i_type.

Reply:
> Can't this be changed in a loop without indexing and where you take the issue type directly from the list, e.g.: for i_type in issue_types: then there is no need for the issue_types[ii] but you can immediately use i_type.

I agree! Thanks

Reply:
Thanks for the input. I have updated the code such that this function is as simple as possible. It now only check for one issue_type.